### PR TITLE
Adds support for gitcoin platform fee

### LINKED ISSUE #3: add support for gitcoin platform fee
per https://medium.com/gitcoin/a-gitcoin-platform-fee-905a0507961f we are now charging a 10% fee on every bounty.

the gitcoin system will not activate the bounty without it.

the scope of this bounty is to add the 10% fee to the standard bounties funder.   the fee (whether ERC20 or ETH) should be paid to 0x00De4B13153673BCAE2616b67bf822500d325Fc3 when the bounty is posted.

### PR DESCRIPTION
**Pull request by pbteja1998**
Resolves #3

### CODE REVIEW

File: funder/utils/token.py [hunk 1]
```diff
@@ -51,6 +51,23 @@ def approveTokenTransfer(state):
     bountiesContract = getBountiesContract(state.get('network'))
     token = getTokenContract(state.get('network'), to_checksum_address(state.get('token_address')))
 
+    # Pay Platform Fees
+    platform_fees_tx = token.functions.approve(
```
Comment (owocki) [inline funder/utils/token.py:57]: what if it's ETH (not a token)?

for token transfers, dont we want to `transfer()` not just `approve()` the tokens?

**Comment by owocki** [comment]
how did you test this?  got any rinkeby transactions generated by the script i can inspect?

File: funder/utils/token.py (hunk 1 above)
Comment (pbteja1998) [reply funder/utils/token.py:57]: @owocki I changed the code, please check now.

**Comment by pbteja1998** [comment]
@owocki  I am unable to test this. I am getting `429 Client Error: Too Many Requests for url: https://rinkeby.infura.io/`. Only one of the transactions(first transaction) is going through.

**Comment by owocki** [comment]
hmmm i wonder if we should add a sleep or retry in this case..

File: funder/commands/issue_and_activate.py [hunk 2]
```diff
@@ -56,13 +93,16 @@ def issueAndActivateBounty(state, ipfsHash):
 def canUserFundBounty(state, ether, tokens=0):
     # check if user has enough ether
     # need to account for gas limit regardless, only add amount if bounty is funded using ether
-    amount = state.get('gas_limit')
+    # need to do two transactions, one to the bountiesNetwork and other as platform fees to gitcoin
+    amount = 2 * state.get('gas_limit')
```
Comment (owocki) [inline funder/commands/issue_and_activate.py]: whats the deal with the `2 *`

**Comment by owocki** [comment]
were you able to get a bounty posted to rinkeby with this code?  if so whats the standard bounties ID of it?

File: funder/commands/issue_and_activate.py (hunk 2 above)
Comment (pbteja1998) [reply funder/commands/issue_and_activate.py]: I thought that gas for each trx is being accounted for, but just saw that it is gas_limit. Will correct this.

**Comment by pbteja1998** [comment]
@owocki Check [this](https://gitcoin.co/issue/pbteja1998/OpenRank/3/2372). Trx with eth went through. Testing with tokens is throwing the error that I mentioned before. I am trying to make it work with sleep and retries.

**Comment by pbteja1998** [comment]
@owocki Even with retries, it is failing. Only the first of all the transactions is going through. 

Can you try from your system once?

**Comment by pbteja1998** [comment]
@owocki What's the status of this ?

**Comment by owocki** [comment]
it needs to be tested/working to get merged

**Comment by pbteja1998** [comment]
@owocki 
Unfortunately, I was unable to get it to work. I am getting `429 Client Error: Too Many Requests for url:` even after introducing retries and sleep. I stopped work on this in gitcoin. 

Thank you.

**Comment by owocki** [comment]
shoot; maybe its worth signing up for an infura plus account @pbteja1998 ?  or asking them how to get around that?  id hate to have you give up the bounty..

maybe if someone else can figure out whats wrong and they use your code you guys can split the bounty..

**Comment by pbteja1998** [comment]
@owocki I will try once with infura plus account. I will update you how it goes.

**Comment by pbteja1998** [comment]
@owocki It worked. https://rinkeby.etherscan.io/tx/0xccef7c9a01a9aa7adf17d959d96b0ed8264ae2dcbf7088359450bdc7a893e844
https://rinkeby.etherscan.io/tx/0xbe3634b55a6767874fe31d4568467de6134d6b5366bb431ed1e28baaf193706c

I don't have any ERC20 tokens to test with. But with ETH, it works, but it is not being shown on gitcoin.

**Comment by pbteja1998** [comment]
@owocki I finally got a hold of some rinkeby ERC20 tokens and tested it.
It worked.

https://rinkeby.etherscan.io/tx/0xe531dc6da47114e9a887edc63742b60b925286956440f220c432319603fe8f5b

https://rinkeby.etherscan.io/tx/0x2a02477577457489323a60b9fc587bb261680f34ef02ff6616749fdfd35472d0

**Comment by pbteja1998** [comment]
**Bounty with ETH**

Sending platform fees to gitcoin... `0x4642e22417c7dbc322e83d3c8a051944ccaa1ad1003352558f87524364a8b0eb`

Funding bounty... 
`0x87d7c0a86febc7ff8dfd213d84fa042410e6ea944f9d6856b6dcd7ba596a1d30`

---

**Bounty with ERC20**

Approving token usage to pay platform fees to gitcoin... 
`0x6ea6735af59089c77979043d02179356d620682c566eec430f6984204b211cf3`

Approving token usage to fund bounty... `0x0d97ef5a0c5b96b84f0773923288dafc36d319cc83969a5947caf168c9962942`

Sending platform fees to gitcoin... `0x671dbea1b19471f6c6e8105b7a06341898ef65dd9964be705de353aea47ab7c1`

Funding bounty... 
`0xcc55002532cba1702b14c47cacca3d55f9eb8b1ffc0efcf3b894e10aa3d694eb`

**Comment by pbteja1998** [comment]
@owocki I used FREE plan of infura which can do 100,000 requests/day. You have to change `project_id` in ` funder/config/web3.py` if you want to have more requests/day.

**Comment by owocki** [comment]
im glad you stuck with it!!!

@pbteja1998 can you add info about how to set the project id in the readme so that whomever uses this in the future/reads the docs can know how to do it!??

**Comment by pbteja1998** [comment]
@owocki I added the relevant instructions to the README

**Comment by pbteja1998** [comment]
@owocki Any more changes that needs to be made?

**Comment by owocki** [comment]
naw; paying out

**Pull request merged**